sdk/tables/azure-data-tables/tests/test_encoder.py

```diff
@@ -539,68 +940,58 @@ def test_encoder_create_entity_atypical_values(
-        )
-        client.create_table()
-        try:
+        ) as client:
+            client.create_table()
             # Non-UTF8 characters in both keys and properties
             test_entity = {"PartitionKey": "PK", "RowKey": "你好", "Data": "你好"}
             expected_entity = {
                 "PartitionKey": "PK",
-                "PartitionKey@odata.type": "Edm.String",
                 "RowKey": "你好",
-                "RowKey@odata.type": "Edm.String",
-                "Data": "你好",
-                "Data@odata.type": "Edm.String",
+                "Data": "你好"
             }
-            _check_backcompat(test_entity, expected_entity)
+            # _check_backcompat(test_entity, expected_entity) # this will fail, as odata type were added for pk and rk
             resp = client.create_entity(
                 test_entity,
-                verify_payload=json.dumps(expected_entity),
+                verify_payload=json.dumps(expected_entity, sort_keys=True),
                 verify_url=f"/{table_name}",
                 verify_headers={"Content-Type": "application/json;odata=nometadata"},
                 verify_response=(lambda: client.get_entity("PK", "你好"), test_entity),
             )
             assert list(resp.keys()) == ["date", "etag", "version"]
 
-            # Invalid int32 and int64 values
-            # TODO: Check with other languages whether they can support big int32. Also Cosmos.
-            # TODO: This will likely change if we move to post-request validation.
+            # # Invalid int32 and int64 values
+            # # TODO: Check whether other languages support big int32. Also Cosmos.
+            # # TODO: This will likely change if we move to post-request validation.
             max_int64 = 9223372036854775808
             test_entity = {"PartitionKey": "PK1", "RowKey": "RK1", "Data": int(max_int64 * 1000)}
             expected_entity = {
                 "PartitionKey": "PK1",
-                "PartitionKey@odata.type": "Edm.String",
                 "RowKey": "RK1",
-                "RowKey@odata.type": "Edm.String",
-                "Data": int(max_int64 * 1000),
+                "Data": int(max_int64 * 1000)
             }
-            with pytest.raises(TypeError):
-                _check_backcompat(test_entity, expected_entity)
-            with pytest.raises(TypeError):
-                resp = client.create_entity(
-                    test_entity,
-                    verify_payload=json.dumps(expected_entity),
-                    verify_url=f"/{table_name}",
-                    verify_headers={"Content-Type": "application/json;odata=nometadata"},
-                    verify_response=(lambda: client.get_entity("PK1", "RK1"), test_entity),
-                )
-                assert list(resp.keys()) == ["date", "etag", "version"]
+            # _check_backcompat(test_entity, expected_entity) # this will fail, as odata type were added for pk and rk
```

Comment:
This should pass now?

Reply:
No, old encoder would throw `TypeError` while doing value range validation.